### PR TITLE
Dont fail error suites

### PR DESCRIPTION
**Pull request by ladisgin**
Before all generated error tests should be fail than run.

```cpp
TEST(error, buggy_function2_test_2)
{
    write_to_file("../../../tests/lib/A", "\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0");
    write_to_file("../../../tests/lib/B", "\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0");
    write_to_file("../../../tests/lib/C", "\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0");

    // Construct input
    int a = 0;

    // Trigger the function
    buggy_function2(a);

    FAIL() << "Unreachable point or the function was supposed to fail, but "
        "actually completed successfully. See the SARIF report for details.";
}
```

In this PR adds the ability to choose whether the test will fail or pass.

image

This use `EXPECT_ANY_THROW` and `ASSERT_DEATH` gtest macros

```cpp

TEST(error, buggy_function2_test_2)
{
    write_to_file("../../../tests/lib/A", "\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0");
    write_to_file("../../../tests/lib/B", "\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0");
    write_to_file("../../../tests/lib/C", "\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0""\0");

    // Construct input
    int a = 0;

    // Trigger the function
    ASSERT_DEATH(buggy_function2(a), ".*");
}
```

Add VSCode setting description at [wiki](https://github.com/UnitTestBot/UTBotCpp/wiki/vscode-extension-settings#error-mode)

### CODE REVIEW

File: server/proto/testgen.proto [hunk 1]
```diff
@@ -90,13 +90,20 @@ message ProjectContext {
     string buildDirRelativePath = 4;
 }
 
+enum ErrorMode {
+    FAILING = 0;
+    PASSING_IN_TARGET_ONLY = 1;
+    PASSING = 2;
+}
```
Comment (Lana243) [inline server/proto/testgen.proto:97]: Could you please add some docs about error mode? Probably, comments or doc at wiki?

File: server/src/KleeGenerator.h [hunk 2]
```diff
@@ -117,7 +117,8 @@ class KleeGenerator {
         const std::unordered_map<std::string, types::Type> &methodNameToReturnTypeMap,
         const std::vector<MethodKtests> &kleeOutput,
         const std::shared_ptr<LineInfo> &lineInfo = nullptr,
-        bool verbose = false);
+                           bool verbose = false,
+                           ErrorMode errorMode = ErrorMode::FAILING);
```
Comment (Lana243) [inline server/src/KleeGenerator.h:121]: Seems like smth wrong with stubs

File: server/src/utils/ErrorInfo.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#define UNITTESTBOT_ERRORINFO_H
+#include <string>
+#include <optional>
+#include "path/FileSystemPath.h"
+
+enum ErrorType {
+    NO_ERROR,
+    ASSERTION_FAILURE,
+    EXCEPTION_THROWN
+};
```
Comment (Lana243) [inline server/src/utils/ErrorInfo.h:11]: Could you please add some docs about error mode? Probably, comments or doc at wiki?

File: server/src/utils/TraceUtils.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+//    ErrorInfo errorInfo;
+//    if (Paths::errorFileExists(path, "assert")) {
+//        errorInfo.errorType = ErrorType::ASSERTION_FAILURE;
+//        fs::path errorFilePath = Paths::replaceExtension(path, ".assert.err");
+//        errorInfo.assertBody = getTypeOfError(errorFilePath);
+//        errorInfo.fileWithFailure = getFileErrorFound(errorFilePath);
+//        errorInfo.lineWithFailure = getLineErrorFound(errorFilePath);
+//    }
+//    return errorInfo;
+//}
```
Comment (Lana243) [inline server/src/utils/TraceUtils.cpp:79]: Why does this piece of code is commented out? If unnecessary, please, remove

**Review by Lana243 — APPROVED** [review]
It would be nice to have some documentation about this PR. How does the behavior of running error test will change after this PR? 
I would recommend to add this information to wiki or to the description of this PR and also comment new entities like ErrorMode and ErrorType in code.

**Review by sava-cska — APPROVED** [review]

**Pull request merged**